app/services/shopify_import/products/create.rb

```diff
@@ -30,6 +31,31 @@ def add_tags
         @spree_product.save!
       end
 
+      def add_option_types
+        return if shopify_product.options.blank?
+
+        @spree_product.update!(option_type_ids: create_option_types)
+      end
+
+      def create_option_types
+        option_types_data.map do |option_type, option_values|
+          spree_option_type = Spree::OptionType.where(name: option_type).first_or_create!(presentation: option_type)
```

Comment:
You're using a lot of bang methods. These could have bad side-effects when one of these fails, for example, when validation fails.